test/spec/proxySpec.js

```diff
@@ -478,7 +478,25 @@ describe('proxy', function() {
 
       observedObj.lastName = 'Wester';
 
-      expect(console.warn).toHaveBeenCalledWith('JSONPatcherProxy noticed a non-integer prop was set for an array. This will not emit a patch');
+      expect(console.warn).toHaveBeenCalledWith("JSONPatcherProxy noticed a non-integer property ('lastName') was set for an array. This interception will not emit a patch");
+    });
+
+    it('should not proxify an object that is assigned as an array prop - and log a warning', function() {
+      var obj = [];
+      var jsonPatcherProxy = new JSONPatcherProxy(obj);
+      var observedObj = jsonPatcherProxy.observe(true);
+
+      spyOn(console, 'warn');
+
+      observedObj.person = {
+        name: "Albert"
+      };
+      observedObj.person.name = "Joachim";
+
+      expect(console.warn).toHaveBeenCalledWith("JSONPatcherProxy noticed a non-integer property ('person') was set for an array. This interception will not emit a patch. The value of this property is an object, but it was not proxified, because only arrays only can have numeric properties in JSON-Pointer");
+
+      var patches = jsonPatcherProxy.generate();
```

Comment:
cosmetic: all `var`s above could be `const`s

Reply:
Also, I think there is too many "only"s;) in "because only arrays only can have numeric properties in JSON-Pointer" (-> "because, in JSON-Pointer, arrays can have only numeric properties"?)

Reply:
> cosmetic: all `var`s above could be `const`s

I will do it just for the test created in this PR. For other tests, I have started a separate PR https://github.com/Palindrom/JSONPatcherProxy/pull/49